numalogic/tools/data.py

```diff
@@ -33,31 +53,59 @@ def __init__(self, data: NDArray[float], seq_len: int):
         self._seq_len = seq_len
         self._data = data.astype(np.float32)
 
-    def create_seq(self, x):
+    def create_seq(self, input_: npt.NDArray[float]) -> Generator[npt.NDArray[float], None, None]:
+        r"""
+        A generator function that yields sequences of specified length from the input data.
+        Yields a subarray from the input data of size (seq_len, num_features)
+
+        Args:
+            input_: A numpy array containing the input data.
+        """
         idx = 0
         while idx < len(self._data) - self._seq_len + 1:
-            yield x[idx : idx + self._seq_len]
+            yield input_[idx : idx + self._seq_len]
             idx += 1
 
-    def __iter__(self):
+    def __iter__(self) -> Iterator[npt.NDArray[float]]:
+        r"""
+        Returns an iterator for the StreamingDataset object.
+
         # TODO implement multi worker iter
+        """
         return iter(self.create_seq(self._data))
 
-    def __len__(self):
+    def __len__(self) -> int:
```

Comment:
nitpick: it appears that you could use this function in all places of `len(self._data) - self._seq_len + 1`